### DISCUSSION TITLE
Uncaught (in promise) TypeError: api.now is not a function

### DISCUSSION THREAD

**Discussion by procrates** [nuxt/nuxt#16750] [Questions] [answered]
Running a Nuxt 3 app with pinia and tailwind.

After upgrading Nuxt 3 i got this error.

```
Uncaught (in promise) TypeError: api.now is not a function
    at vue-router.esm-bundler.js?v=44d7056b:2545:31
    at triggerAfterEach (vue-router.esm-bundler.js?v=44d7056b:3166:13)
    at vue-router.esm-bundler.js?v=44d7056b:3069:13
```
Line where the error occures:
```
api.addTimelineEvent({
                layerId: navigationsLayerId,
                event: {
                    title: 'End of navigation',
                    subtitle: to.fullPath,
                    time: api.now(), <--
                    data,
                    logType: failure ? 'warning' : 'default',
                    groupId: to.meta.__navigationId,
                },
            }); 
```

Project info: 
```
------------------------------
- Operating System: `Darwin`
- Node Version:     `v17.5.0`
- Nuxt Version:     `3.0.0-27470397.9ebea90`
- Package Manager:  `npm@8.4.1`
- Builder:          `vite`
- User Config:      `privateRuntimeConfig`, `buildModules`, `build`, `postcss`, `vite`
- Runtime Modules:  `-`
- Build Modules:    `@pinia/nuxt@0.1.8`
------------------------------
```

**Comment by Diizzayy** [comment, marked as answer]
@procrates That error along with the code you referenced are from `vue-router` vuejs/router#1338

As for why you've encountered this bug using nuxt3, can you provide more details or a reproduction?

**Reply by procrates** [reply]
Wow... That fixed it. Was so strange since the bug suddenly appeared few weeks ago.. Thank you very much!

**Reply by droidzone** [reply]
What fixed it?

**Reply by Pierpaolo01** [reply]
I'm getting this error aswell with vue3 and vue-router-4. Really wierd bug, can we get some insights in what fixed it?

**Reply by steswinbank** [reply]
Old version of Chrome Vue Devtools was causing this. I was still using the Beta version.

**Reply by nidhi-canopas** [reply]
It worked for me as well. Thanks !

**Reply by socraticprogrammer** [reply]
Using the latest version of [Vue DevTools](https://chrome.google.com/webstore/detail/vuejs-devtools/ljjemllljcmogpfapbkkighbhhppjdbg) worked for me.

**Reply by yegormoss** [reply]
This issue is caused when you install Vue router 3 with Vue 3 so you should uninstall the current version :

npm uninstall vue-router --save

and install the new one by :

npm i vue-router@next --save

**Reply by Morriso8D** [reply]
Annoying bug. Switching from the beta Vue Devtools to stable seemed to do the trick. Thanks all

**Reply by radislaw** [reply]
now this bug appears after installing vue dev tools beta

**Reply by Jiancong** [reply]
uninstall the vue-router and reinstall the latest version, works for me.

**Comment by yegormoss** [comment]
This issue is caused when you install Vue router 3 with Vue 3 so you should uninstall the current version :

npm uninstall vue-router --save

and install the new one by :

npm i vue-router@next --save

**Reply by wil-ixcoy** [reply]
Thank you

**Reply by montalvoalmag** [reply]
Thank you, Gracias.

**Reply by Wanhsuan98** [reply]
Thank you!!!! You helped me a lot :))

**Reply by Shohzod-Abdusamatov-7777777** [reply]
Thank you  👍

**Reply by johnwangxtu** [reply]
Thank You!

**Reply by TiagoSilvaPereira** [reply]
Thank you, it worked here

**Reply by bigCows** [reply]
thanks，my bro
